packages/contracts-bedrock/src/L1/SystemConfigHolocene.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { OptimismPortalInterop as OptimismPortal } from "src/L1/OptimismPortalInterop.sol";
+import { SystemConfigInterop } from "src/L1/SystemConfigInterop.sol";
+import { ConfigType } from "src/L2/L1BlockInterop.sol";
+import { StaticConfig } from "src/libraries/StaticConfig.sol";
+
+/// @title SystemConfigHolocene
+/// @notice The SystemConfig contract is used to manage configuration of an Optimism network.
+///         All configuration is stored on L1 and picked up by L2 as part of the derviation of
+///         the L2 chain.
+contract SystemConfigHolocene is SystemConfigInterop {
+    /// @notice Internal setter for the batcher hash.
+    /// @param _batcherHash New batcher hash.
+    function _setBatcherHash(bytes32 _batcherHash) internal override {
+        if (batcherHash != _batcherHash) {
+            batcherHash = _batcherHash;
+            OptimismPortal(payable(optimismPortal())).setConfig(
+                ConfigType.SET_BATCHER_HASH, StaticConfig.encodeSetBatcherHash({ _batcherHash: _batcherHash })
+            );
+        }
+    }
+
+    /// @notice Internal function for updating the fee scalars as of the Ecotone upgrade.
+    /// @param _basefeeScalar     New basefeeScalar value.
+    /// @param _blobbasefeeScalar New blobbasefeeScalar value.
+    function _setGasConfigEcotone(uint32 _basefeeScalar, uint32 _blobbasefeeScalar) internal override {
```

Comment:
Mark as deprecated, and the implementation here can just call:

_setBasefeeScalar(_basefeeScalar);
_setBlobBasefeeScalar(_blobBasefeeScalar);

Reply:
Note: This approach would result in _setBasefeeScalar and _setBlobBasefeeScalar each emitting a separate event.
There's a trade-off here between readability and gas efficiency. Are we in agreement that we prioritize readability over gas savings in this instance?